app/scripts/controllers/necesidad/solicitud_necesidad.js

```diff
@@ -183,36 +184,26 @@ angular
             self.dependencia_destino = response.dependencia.Id;
           }): _;
 
-          self.Necesidad.DependenciaNecesidadId.OrdenadorGastoId
-            ? necesidadService
-                .get_info_dependencia(
-                  self.Necesidad.DependenciaNecesidadId.OrdenadorGastoId
-                )
-                .then(function (response) {
-                  self.rol_ordenador_gasto = response.dependencia.Id;
-                })
-            : _;
-
-          if (self.Necesidad.DependenciaNecesidadId.InterventorId === 0) {
-            //verifica si es supervisor o interventor
+          if (self.Necesidad.DependenciaNecesidadId.OrdenadorGastoId) {
+            necesidadService
+            .get_info_dependencia(self.Necesidad.DependenciaNecesidadId.OrdenadorGastoId)
+            .then(function (response) {
+                self.rol_ordenador_gasto = response.dependencia.Id;
+            });
+          }
+          if (self.Necesidad.DependenciaNecesidadId.InterventorId === 0) {//verifica si es supervisor o interventor
             self.tipoInterventor = false;
-            self.Necesidad.DependenciaNecesidadId.SupervisorId
-              ? necesidadService
-                  .get_info_dependencia(
-                    self.Necesidad.DependenciaNecesidadId.SupervisorId
-                  )
-                  .then(function (response) {
-                    self.dependencia_supervisor = response.dependencia.Id;
-                  })
-              : _;
+            self.Necesidad.DependenciaNecesidadId.SupervisorId ? necesidadService.get_info_dependencia(
+              self.Necesidad.DependenciaNecesidadId.SupervisorId
+            ).then(function (response) {
+              self.dependencia_supervisor = response.dependencia.Id;
+            }): _;
```

Comment:
Si no se necesita el valor retornado por una terna, usar un `if`. En general, reducir/evitar el uso de ternas, entre más explícito y legible sea el código, mejor
```suggestion
            if (self.Necesidad.DependenciaNecesidadId.SupervisorId) {
              necesidadService.get_info_dependencia(
                self.Necesidad.DependenciaNecesidadId.SupervisorId
              ).then(function (response) {
                self.dependencia_supervisor = response.dependencia.Id;
              });
            }
```